test/unittests/typed_value_test.cpp

```diff
@@ -73,3 +73,23 @@ TEST(typed_value, construct)
     EXPECT_EQ(f64.type, ValType::f64);
     EXPECT_EQ(f64.value.f64, -2.2250738585072014e-308);
 }
+
+TEST(typed_value, u32_literal)
+{
+    static_assert(std::is_same_v<decltype(0_u32), uint32_t>);
+    EXPECT_EQ(0_u32, uint32_t{0});
+    EXPECT_EQ(1_u32, uint32_t{1});
```

Comment:
What happens with negative literals?

Reply:
There are no negative literals. The `-1` is `"-" + literal "1"`.